File: sample/androidx/host/src/main/java/club/knifelf/androidx/host/MainActivity.java
package club.knifelf.androidx.host;

import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

import com.qihoo360.replugin.RePlugin;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.btn_open_plugin).setOnClickListener(v -> openPlugin());
        if (getIntent().getBooleanExtra("auto_open", false)) {
            findViewById(R.id.btn_open_plugin).post(this::openPlugin);
        }
    }

    private void openPlugin() {
        boolean ok = RePlugin.startActivity(this,
                RePlugin.createIntent("androidx", "club.knifelf.androidx.plugin.MainActivity"));
        if (!ok) {
            Toast.makeText(this, "plugin androidx not installed", Toast.LENGTH_LONG).show();
        }
    }
}

File: sample/compose/host/src/main/java/club/knifelf/compose/host/MainActivity.kt
package club.knifelf.compose.host

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.qihoo360.replugin.RePlugin

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<android.view.View>(R.id.btn_open_plugin).setOnClickListener { openPlugin() }
        if (intent.getBooleanExtra("auto_open", false)) {
            findViewById<android.view.View>(R.id.btn_open_plugin).post { openPlugin() }
        }
    }

    private fun openPlugin() {
        val ok = RePlugin.startActivity(
            this,
            RePlugin.createIntent("compose", "club.knifelf.compose.plugin.MainActivity")
        )
        if (!ok) {
            Toast.makeText(this, "plugin compose not installed", Toast.LENGTH_LONG).show()
        }
    }
}
